### PR TITLE
ARROW-2181: [PYTHON][DOC] Add doc on usage of concat_tables

### PR DESCRIPTION
**Pull request by BryanCutler**
Adding Python API doc on usage of pa.concat_tables.

### CODE REVIEW

**Comment by BryanCutler** [comment]
cc @wesm @xhochy , I didn't get a chance to build the docs to try this out yet.  Is it done with the gen_api_docs docker image, or is there another easier way to do it?

File: python/doc/source/data.rst [hunk 1]
```diff
@@ -393,6 +393,22 @@ objects to contiguous NumPy arrays for use in pandas:
...
+   tables = [table] * 2
+   table_all = pa.concat_tables(tables)
+   table_all.num_rows
+   c = table_all[0]
+   c.data.num_chunks
+
+This is similar to ``Table.from_batches``, but uses tables as input instead of
+record batches. Record batches can be made into tables, but not the other way
+around, so if your data is already in table form, then use
+``pa.concat_tables``.
```
Comment (wesm) [inline python/doc/source/data.rst]: Can you spell out `pyarrow` here and above? We might turn these into API links later

**Comment by wesm** [comment]
see https://github.com/apache/arrow/tree/master/python#building-the-documentation -- if you are able to run the test suite / build the project then you can run those commands

File: python/doc/source/data.rst (hunk 1 above)
Comment (BryanCutler) [reply python/doc/source/data.rst]: Sure, will do

**Comment by BryanCutler** [comment]
Great thanks, I'll run it

**Comment by BryanCutler** [comment]
Screen of doc changes
image

**Review by wesm — APPROVED** [review]
+1, thanks @BryanCutler!

**Pull request closed without merging**